overlord/devicestate/remodel_test.go

```diff
@@ -0,0 +1,710 @@
+		Brand:           "my-brand",
+		Model:           "my-model",
+		Serial:          "serialserialserial",
+		SessionMacaroon: "new-session",
+	})
+	c.Assert(err, IsNil)
+
+	err = remodCtx.Finish()
+	c.Assert(err, IsNil)
+
+	// the global device now matches the state reached in the remodel
+	expectedGlobalDevice := &auth.DeviceState{
+		Brand:           "my-brand",
+		Model:           "my-model",
+		Serial:          "serialserialserial",
+		SessionMacaroon: "new-session",
+	}
+
+	device, err := s.mgr.StoreContextBackend().Device()
+	c.Assert(err, IsNil)
+	c.Check(device, DeepEquals, expectedGlobalDevice)
+}
+
+func (s *remodelLogicSuite) TestNewStoreRemodelContextFinishVsGlobalUpdateDeviceAuth(c *C) {
+	oldModel := fakeRemodelingModel(nil)
+	newModel := fakeRemodelingModel(map[string]interface{}{
+		"store":    "my-other-store",
+		"revision": "1",
+	})
+
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	// we have a device state
+	devicestatetest.SetDevice(s.state, &auth.DeviceState{
+		Brand:           "my-brand",
+		Model:           "my-model",
+		Serial:          "serialserialserial",
+		SessionMacaroon: "old-session",
+	})
+
+	remodCtx, err := devicestate.RemodelCtx(s.state, oldModel, newModel)
+	c.Assert(err, IsNil)
+
+	chg := s.state.NewChange("remodel", "...")
+
+	err = remodCtx.Init(chg)
+	c.Assert(err, IsNil)
+
+	devBE := devicestate.RemodelDeviceBackend(remodCtx)
+
+	err = devBE.SetDevice(&auth.DeviceState{
+		Brand:           "my-brand",
+		Model:           "my-model",
+		Serial:          "serialserialserial",
+		SessionMacaroon: "remodel-session",
+	})
+	c.Assert(err, IsNil)
+
+	// global store device and auth context
+	scb := s.mgr.StoreContextBackend()
+	dac := storecontext.New(s.state, scb)
+	// this is the unlikely start of the global store trying to
+	// refresh the session
+	s.state.Unlock()
+	globalDevice, err := dac.Device()
+	s.state.Lock()
+	c.Assert(err, IsNil)
+	c.Check(globalDevice.SessionMacaroon, Equals, "old-session")
+
+	err = remodCtx.Finish()
+	c.Assert(err, IsNil)
+
+	s.state.Unlock()
+	device1, err := dac.UpdateDeviceAuth(globalDevice, "fresh-session")
+	s.state.Lock()
+	c.Assert(err, IsNil)
+
+	// the global device now matches the state reached in the remodel
+	expectedGlobalDevice := &auth.DeviceState{
+		Brand:           "my-brand",
+		Model:           "my-model",
+		Serial:          "serialserialserial",
+		SessionMacaroon: "remodel-session",
+	}
+
+	s.state.Unlock()
+	device, err := dac.Device()
+	s.state.Lock()
+	c.Assert(err, IsNil)
+	c.Check(device, DeepEquals, expectedGlobalDevice)
+
+	// also this was already the case
+	c.Check(device1, DeepEquals, expectedGlobalDevice)
+}
+
+func (s *remodelLogicSuite) TestRemodelDeviceBackendSerial(c *C) {
+	oldModel := fakeRemodelingModel(nil)
+	newModel := fakeRemodelingModel(map[string]interface{}{
+		"store":    "my-other-store",
+		"revision": "1",
+	})
+
+	// the logic is shared and correct also for re-reg
+	// XXX (this is really the re-reg case)
```

Comment:
Is the plan to change it stil in this PR or when re-reg is handled (in a followup)?

Reply:
in a follow up